### PR TITLE
Use goembed for all tests

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Comment by mathetake** [comment]
Oh didn't know that goembed can be used for directory so I can use it for spectests as well! lemme

**Comment by mathetake** [comment]
ok ready!

File: tests/spectest/spec_test.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ import (
 	"github.com/tetratelabs/wazero/wasm/jit"
 )
 
+//go:embed testdata/*
```
Comment (codefromthecrypt) [inline tests/spectest/spec_test.go:23]: nit does this accept *.wasm? it will help me in inventory of which tests still use binaries

**Review by codefromthecrypt — APPROVED** [review]
this is great. at least in goland this is tighter as the IDE can tell when you reference a file that doesn't exist. That's as good imho as the reduction of error handling

File: tests/spectest/spec_test.go (hunk 1 above)
Comment (mathetake) [reply tests/spectest/spec_test.go:23]: not really, we have to read *.json first and inside of it wasm binary names are specified.

Comment (mathetake) [reply tests/spectest/spec_test.go]: https://github.com/tetratelabs/wazero/pull/197/commits/58cdd090272be91978c2fef72ceaf40d10c42172 specified at least!

**Pull request merged**